docs/changes/table/12825.feature.rst

```diff
@@ -0,0 +1,4 @@
+Add support for using mixin columns in group aggregation operations when the
+mixin supports the specified operation (e.g. ``np.sum`` works for ``Quantity``
+but not ``Time``). In cases where the operation is not supported the code now
```

Comment:
Indeed I did that check previously and .. it does not work. The problem is that you cannot currently initialize `TimeDelta` from an object array of `TimeDelta` objects, which is what the `out = par_col.__class__(vals)` statement ends up doing. Sigh, so that ends up being a *different* issue which is solvable, but not in this PR.

Reply:
Indeed, sounds like it would need `__array_ufunc__` support in `Time/TimeDelta` - that will take a bit more effort than I have time for right now... (#10272 was start but didn't quite go far enough; bit of an all-or-nothing case).

Anyway, for another PR!